pkg/hintrunner/zero/zerohint_signatures.go

```diff
@@ -50,3 +53,117 @@ func createVerifyECDSASignatureHinter(resolver hintReferenceResolver) (hinter.Hi
+			//> from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+			//> x_cube_int = pack(ids.x_cube, PRIME) % SECP_P
+			//> y_square_int = (x_cube_int + ids.BETA) % SECP_P
+			//> y = pow(y_square_int, (SECP_P + 1) // 4, SECP_P)
+			//> if ids.v % 2 == y % 2:
+			//>	 value = y
+			//> else:
+			//>	 value = (-y) % SECP_P
+
+			xCubeMemoryValues, err := hinter.GetConsecutiveValues(vm, xCube, 3)
+			if err != nil {
+				return err
+			}
+			v, err := hinter.ResolveAsFelt(vm, v)
+			if err != nil {
+				return err
+			}
+
+			//> from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+			secpBig, _ := utils.GetSecPBig()
+
+			//> x_cube_int = pack(ids.x_cube, PRIME) % SECP_P
+			var xCubeValues [3]*fp.Element
+			for i := 0; i < 3; i++ {
+				xCubeValues[i], err = xCubeMemoryValues[i].FieldElement()
+				if err != nil {
+					return err
+				}
+			}
+			xCubeIntBig, err := utils.SecPPacked(xCubeValues)
+			if err != nil {
+				return err
+			}
+			xCubeIntBig.Mod(xCubeIntBig, secpBig)
+
+			//> y_square_int = (x_cube_int + ids.BETA) % SECP_P
+			ySquareIntBig := new(big.Int).Add(xCubeIntBig, utils.GetBetaBig())
+			ySquareIntBig.Mod(ySquareIntBig, secpBig)
+
+			//> y = pow(y_square_int, (SECP_P + 1) // 4, SECP_P)
+			y := new(big.Int).Exp(ySquareIntBig, new(big.Int).Div(new(big.Int).Add(secpBig, big.NewInt(1)), big.NewInt(4)), secpBig)
+			vBig := v.BigInt(new(big.Int))
+
+			//> if ids.v % 2 == y % 2:
+			//>	 value = y
+			//> else:
+			//>	 value = (-y) % SECP_P
+			value := new(big.Int)
+			if vBig.Bit(0) == y.Bit(0) {
+				value.Set(y)
+			} else {
+				value.Mod(value.Neg(y), secpBig)
+			}
+			return ctx.ScopeManager.AssignVariable("value", value)
+		},
+	}
+}
+
+func createGetPointFromXHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
+	xCube, err := resolver.GetResOperander("x_cube")
+	if err != nil {
+		return nil, err
+	}
+	v, err := resolver.GetResOperander("v")
+	if err != nil {
+		return nil, err
+	}
+	return newGetPointFromXHinter(xCube, v), nil
+}
+
+func newDivModSafeDivHinter() hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "DivModSafeDivHinter",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> value = k = safe_div(res * b - a, N)
+
+			res, err := ctx.ScopeManager.GetVariableValueAsBigInt("res")
+			if err != nil {
+				return err
+			}
+			a, err := ctx.ScopeManager.GetVariableValueAsBigInt("a")
+			if err != nil {
+				return err
+			}
+			b, err := ctx.ScopeManager.GetVariableValueAsBigInt("b")
+			if err != nil {
+				return err
+			}
+			N, err := ctx.ScopeManager.GetVariableValueAsBigInt("N")
+			if err != nil {
+				return err
+			}
+			divisor := new(big.Int).Sub(new(big.Int).Mul(res, b), a)
+			value, err := utils.SafeDiv(divisor, N)
+			if err != nil {
+				return err
+			}
+			err = ctx.ScopeManager.AssignVariable("k", value)
+			if err != nil {
+				return err
+			}
+			return ctx.ScopeManager.AssignVariable("value", value)
```

Comment:
Shouldn't k and value point to different big.int objects?

Reply:
Not sure if this is necessary, but added it for safety and readability